classes/ThankYous/ThankYouAcl.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Claromentis\ThankYou\ThankYous;
+
+use Claromentis\Core\Admin\AdminPanel;
+use Claromentis\Core\Security\SecurityContext;
+use Claromentis\People\Service\UserExtranetService;
+use Claromentis\ThankYou\Thankable\Thankable;
+use InvalidArgumentException;
+use User;
+
+class ThankYouAcl
+{
+	/**
+	 * @var AdminPanel
+	 */
+	private $admin_panel;
+
+	/**
+	 * @var UserExtranetService
+	 */
+	private $user_extranet;
+
+	/**
+	 * ThankYouAcl constructor.
+	 *
+	 * @param AdminPanel          $admin_panel
+	 * @param UserExtranetService $user_extranet_service
+	 */
+	public function __construct(AdminPanel $admin_panel, UserExtranetService $user_extranet_service)
+	{
+		$this->admin_panel   = $admin_panel;
+		$this->user_extranet = $user_extranet_service;
+	}
+
+	/**
+	 * Determines whether a Security Context can Delete a Thank You.
+	 *
+	 * @param ThankYou        $thank_you
+	 * @param SecurityContext $context
+	 * @return bool
+	 */
+	public function CanDeleteThankYou(ThankYou $thank_you, SecurityContext $context): bool
+	{
+		return $thank_you->GetAuthor()->GetId() === $context->GetUser()->GetId() || $this->IsAdmin($context);
+	}
+
+	/**
+	 * Determines whether a Security Context can Edit a Thank You.
+	 *
+	 * @param ThankYou        $thank_you
+	 * @param SecurityContext $context
+	 * @return bool
+	 */
+	public function CanEditThankYou(ThankYou $thank_you, SecurityContext $context): bool
```

Comment:
Small thing, but it would be good to have `SecurityContext` consistently the first argument for all methods in this class that use it.

PhpStorm's refactoring tool should allow you change the signature of the method and update all usages automatically.

Reply:
Fixed in 86071730c11aab105a74d2fa8b2be3b585b282a8